datadog_checks_base/tests/base/utils/replay/test_check.py

```diff
@@ -0,0 +1,55 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import logging
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.constants import ServiceCheck
+from datadog_checks.dev.testing import requires_py3
+
+pytestmark = [requires_py3]
+
+
+class ReplayCheck(AgentCheck):
+    __NAMESPACE__ = 'replay'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.redirecting = not self.instance.get('process_isolation', False)
+        self.tags = ['redirecting:{}'.format(str(self.redirecting).lower())]
+        self.tags.extend(self.instance.get('tags', []))
+        self.tags.extend(self.init_config.get('tags', []))
+
+        self.check_initializations.append(self.initialize)
+
+    def initialize(self):
+        self.gauge('initialize', 0, tags=self.tags)
+        self.log.debug('Initializing - %s - %s', self.name, self.check_id)
+
+    def check(self, _):
+        self.gauge('metric', 0, tags=self.tags)
+        self.service_check('sc', ServiceCheck.OK if self.redirecting else ServiceCheck.CRITICAL, tags=self.tags)
+
+
+def test_replay_all(caplog, dd_run_check, aggregator, datadog_agent):
+    datadog_agent._config['log_level'] = 'debug'
+
+    check = ReplayCheck('replay', {'tags': ['bar:baz']}, [{'process_isolation': True, 'tags': ['foo:bar']}])
+    check.check_id = 'test:123'
+
+    with caplog.at_level(logging.DEBUG):
+        dd_run_check(check)
+
+    expected_tags = ['redirecting:true', 'bar:baz', 'foo:bar']
+
+    aggregator.assert_metric('replay.initialize', 0, count=1, tags=expected_tags)
+    aggregator.assert_metric('replay.metric', 0, count=1, tags=expected_tags)
+    aggregator.assert_service_check('replay.sc', ServiceCheck.OK, count=1, tags=expected_tags)
+
```

Comment:
nit- can we add `assert_all_metrics_covered`?

Reply:
Done!